modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/ExportImportObjectDefinitionTest.java

```diff
@@ -194,27 +243,33 @@ private ThemeDisplay _getThemeDisplay() throws Exception {
 		return themeDisplay;
 	}
 
-	private ObjectDefinition _importObjectDefinition() throws Exception {
+	private ObjectDefinition _importCustomObjectDefinition(
+			String fileName, String name,
+			ObjectDefinitionResource objectDefinitionResource)
+		throws Exception {
+
 		_mvcActionCommand.processAction(
-			_createMockLiferayPortletActionRequest(
-				"object_definition.json", "ImportedObjectDefinition"),
+			_createMockLiferayPortletActionRequest("", fileName, name),
 			new MockLiferayPortletActionResponse());
 
-		ObjectDefinitionResource.Builder builder =
-			_objectDefinitionResourceFactory.create();
+		return _getObjectDefinitionByName(name, objectDefinitionResource);
+	}
 
-		ObjectDefinitionResource objectDefinitionResource = builder.user(
-			TestPropsValues.getUser()
-		).build();
+	private ObjectDefinition _importSystemObjectDefinition(
+			String fileName, String name,
+			ObjectDefinitionResource objectDefinitionResource)
+		throws Exception {
 
-		Page<ObjectDefinition> page =
-			objectDefinitionResource.getObjectDefinitionsPage(
-				"ImportedObjectDefinition", null, null, Pagination.of(1, 1),
-				null);
+		ObjectDefinition objectDefinition = _getObjectDefinitionByName(
+			name, objectDefinitionResource);
 
-		List<ObjectDefinition> items = (List<ObjectDefinition>)page.getItems();
+		_mvcActionCommand.processAction(
+			_createMockLiferayPortletActionRequest(
+				objectDefinition.getExternalReferenceCode(), fileName, name),
+			new MockLiferayPortletActionResponse());
 
-		return items.get(0);
+		return objectDefinitionResource.getObjectDefinition(
+			objectDefinition.getId());
```

Comment:
@pedro-oliveira446 I believe we can join the methods _importCustomObjectDefinition and _importSystemObjectDefinition in only one by passing another parameter, `boolean system`,

could you please try to do it?

Thanks!